src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs

```diff
@@ -68,5 +68,33 @@ public T Get<T>(string id, DateTime date)
             operation.SetResult(command.Result);
             return operation.GetRevisionsFor<T>().FirstOrDefault();
         }
+        
+        public void ForceRevisionCreationFor<T>(T entity, ForceRevisionStrategy forceRevisionCreationStrategy = ForceRevisionStrategy.Before)
+        {
+            if (ReferenceEquals(entity, null))
+                throw new ArgumentNullException(nameof(entity));
+           
+            if (DocumentsByEntity.TryGetValue(entity, out DocumentInfo documentInfo) == false)
+            {
+                throw new InvalidOperationException("Cannot create a revision for the requested entity because it is not tracked by the session");
+            }
+
+            if (Session.IDsForCreatingForcedRevisions.ContainsKey(documentInfo.Id))
+            {
```

Comment:
If they are doing the same operation, why throw? Can we just check that the same `ForceRevisionStrategy` is applied and treat is a noop?

Reply:
I think we need to throw, because we will explicitly inform user how this works. If we will not throw then his intention might be to create N-number of revisions.

Reply:
I think that this is more likely that he might be trying to register the same thing twice.
Once he is doing that by id, once with the entity (can happen in complex cases).
That isn't an _error_, the end result is just the same.
And we don't even have a way for him to _check_ that.

Reply:
Done. Now throwing only if a different strategy is requested in the same session.